src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -245,17 +188,26 @@ async def mypy_typecheck_partition(partition: MyPyPartition, mypy: MyPy) -> Type
             internal_only=True,
         ),
     )
-    mypy_pex_request = Get(
+    # TODO(John Sirois): Scope the extra requirements to the partition.
```

Comment:
To be clear - this was an existing bug. I just noticed it and am calling the shot.